crates/aptos-rosetta/src/block.rs

```diff
@@ -78,42 +56,48 @@ async fn block(request: BlockRequest, server_context: RosettaContext) -> ApiResu
             let response = rest_client.get_transaction(hash).await?;
             let txn = response.into_inner();
             let version = txn.version().unwrap();
+            let block_index = version_to_block_index(server_context.block_size, version);
 
-            // If this is genesis, set parent to genesis txn
-            if version == 0 {
-                (txn.clone(), txn)
-            } else {
-                let parent_response = rest_client.get_transaction_by_version(version - 1).await?;
-                (parent_response.into_inner(), txn)
-            }
+            get_block_by_index(rest_client, server_context.block_size, block_index).await?
         }
         (None, None) => {
             // Get current version
-            let response = rest_client.get_transactions(None, Some(2)).await?;
-            let txns = response.into_inner();
-            if txns.len() != 2 {
-                return Err(ApiError::AptosError(
-                    "Failed to get transaction and parent transaction".to_string(),
-                ));
-            }
-            (
-                txns.first().cloned().unwrap(),
-                txns.last().cloned().unwrap(),
-            )
+            let response = rest_client.get_ledger_information().await?;
+            let version = response.state().version;
+            let block_index = version_to_block_index(server_context.block_size, version) - 1;
+
+            get_block_by_index(rest_client, server_context.block_size, block_index).await?
         }
         (_, _) => return Err(ApiError::BadBlockRequest),
     };
 
     // Build up the transaction, which should contain the `operations` as the change set
-    let transaction_info = transaction.transaction_info()?;
-    let transactions = vec![transaction_info.into()];
+    let (block_identifier, timestamp) = if let Some(first) = transactions.first() {
+        // note: timestamps are in microseconds, so we convert to milliseconds
+        let mut timestamp = first.timestamp() / 1000;
 
-    // note: timestamps are in microseconds, so we convert to milliseconds
-    let timestamp = transaction.timestamp() / 1000;
+        // Rosetta doesn't like timestamps before 2000
+        if timestamp == 0 {
+            timestamp = 946713600000;
```

Comment:
maybe we can put this into something readable and then parse it out so we know what this represents or you could write it out? Also it seems like this should be if timestamp < 2000 not just == 0

Reply:
Yeah, let me make it a constant.